packages/headless/src/database/mongodb/product/index.ts

```diff
@@ -1,23 +1,20 @@
+import { Injectable } from '@nestjs/common';
 import { Product } from '../../../entity/product';
 import { IProductDatabase } from '../../../modules/product/repositories/product.database.interface';
 import { ProductModel } from './product.model';
-import { Injectable } from '@nestjs/common';
 
 @Injectable()
 export class ProductDatabase implements IProductDatabase {
-    constructor() { }
-
-    async findById(productId: string) {
-        const product = await ProductModel.findOne({ id: productId });
-        return Promise.resolve(product);
-    }
-    async save(product: Product) {
-        await ProductModel.create(product);
-        return Promise.resolve(product);
-    }
+  async findById(productId: string) {
+    return await ProductModel.findOne({ id: productId }).lean();
+  }
+  async save(product: Product) {
+    await ProductModel.create(product);
+    return Promise.resolve(product);
```

Comment:
here you can also do the same as previous function. returning the result in the same line where you are querying. You can also keep this as it is. Because this is not a major issue. But in future you can use single line code.
This is not mandatory to do now, but a suggestion.